dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Basics/SwitchOff/SwitchOffLogic.mo

```diff
@@ -14,15 +14,20 @@ within Dynawo.Electrical.Controls.Basics.SwitchOff;
 */
 
 partial model SwitchOffLogic "Manage switch-off logic"
-  /* Handles a predefinite number of switch-off signals and sets running to false as soon as one signal is set to true */
 
-  Dynawo.Connectors.BPin switchOffSignal1(value(start = false)) "Switch-off signal 1";
-  Dynawo.Connectors.BPin switchOffSignal2(value(start = false)) if NbSwitchOffSignals >= 2 "Switch-off signal 2";
-  Dynawo.Connectors.BPin switchOffSignal3(value(start = false)) if NbSwitchOffSignals >= 3 "Switch-off signal 3";
+  parameter Integer NbSwitchOffSignals(min = 1, max = 3) "Number of switch-off signals to take into account in inputs";
 
-  Dynawo.Connectors.BPin running(value(start = true)) "Indicates if the component is running or not";
+  Dynawo.Connectors.BPin switchOffSignal1(value(start = SwitchOffSignal10)) "Switch-off signal 1";
+  Dynawo.Connectors.BPin switchOffSignal2(value(start = SwitchOffSignal20)) if NbSwitchOffSignals >= 2 "Switch-off signal 2";
+  Dynawo.Connectors.BPin switchOffSignal3(value(start = SwitchOffSignal30)) if NbSwitchOffSignals >= 3 "Switch-off signal 3";
 
-  parameter Integer NbSwitchOffSignals(min = 1, max = 3) "Number of switch-off signals to take into account in inputs";
+  Dynawo.Connectors.BPin running(value(start = Running0)) "Indicates if the component is running or not";
+
+  parameter Boolean SwitchOffSignal10 = false "Initial switch-off signal 1";
+  parameter Boolean SwitchOffSignal20 = false "Initial switch-off signal 2";
+  parameter Boolean SwitchOffSignal30 = false "Initial switch-off signal 3";
+
+  final parameter Boolean Running0 = not(SwitchOffSignal10 or SwitchOffSignal20 or SwitchOffSignal30) "Indicates if the component is initially running or not";
```

Comment:
But maybe we can add the if condition here as it was previously, just to keep Running0 dependent on the NbSwitchOffSignals (for example, if the user accidentally added a value for switchoff30 when we have NbSwitchOffSignals < 3).

Reply:
Running0 now depends on NbSwitchOffSignals. All nrt have passed.